retrodep/gosource.go

```diff
@@ -533,19 +533,23 @@ func (src GoSource) RepoPathForImportPath(importPath string) (*RepoPath, error)
 
 	// No replacement found, use the import pth as-is
 	r, err := vcs.RepoRootForImportPath(importPath, false)
-	if err != nil && strings.ContainsRune(importPath, '_') {
-		// gopkg.in gives bad responses for paths like
-		// gopkg.in/foo/bar.v2/_examples/chat1
-		// because of the underscore. Remove it and try again.
-		u := strings.Index(importPath, "_")
-		importPath = path.Dir(importPath[:u])
-		r, nerr := vcs.RepoRootForImportPath(importPath, false)
-		if nerr == nil {
-			return &RepoPath{RepoRoot: *r}, nil
-		}
+	if err == nil {
+		return &RepoPath{RepoRoot: *r}, err
 	}
 
-	return &RepoPath{RepoRoot: *r}, err
+	if !strings.ContainsRune(importPath, '_') {
+		return nil, err
+	}
+	// gopkg.in gives bad responses for paths like
+	// gopkg.in/foo/bar.v2/_examples/chat1
+	// because of the underscore. Remove it and try again.
+	u := strings.Index(importPath, "_")
```

Comment:
This is a bit confused, and I realise the _before_ version was too rather than it being this commit's doing.
Underscore is searched for twice when once would do, and the first time it's called a rune, the second a string but we know it's a byte.

    u := strings.IndexByte(importPath, '_')
    if u == -1 {
        return ...

Reply:
Done